src/libsync/creds/oauth.cpp

```diff
@@ -224,16 +224,22 @@ void OAuth::startAuthentication()
                     httpReplyAndClose(socket, QByteArrayLiteral("400 Bad Request"), QByteArrayLiteral("<html><head><title>400 Bad Request</title></head><body><center><h1>400 Bad Request</h1></center></body></html>"));
                     return;
                 }
+
+                // server port cannot be queried any more after server has been closed, which we want to do as early as possible in the processing chain
+                // therefore we have to store it beforehand
+                const auto serverPort = _server.serverPort();
+
                 // we only allow one response
-                qCDebug(lcOauth) << "Recieved the first valid request, stoping to listen";
+                qCDebug(lcOauth) << "Received the first valid response, closing server socket";
```

Comment:
Actually we received a GET request as a response :P so wording is complicated here.